File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

//declaring variables for change, coins and coin count
float change;
float quarter = 25;
float dime = 10;
float nickel = 5;
float penny = 1;
int count = 0;

int main(void)
{
    do
    {
        change = get_float("Enter change: ");
    }
    while (change < 0); //if change is less than 0 then repeat prompt

    change *= 100; //converting my float to an int to avoid tiny errors along the decimal place
    change = round(change); //round the change to the decimal place

    while (change >= quarter) //check if the quarter can be used
    {
        count++;
        change -= quarter;
    }

    while (change >= dime) //check if the dime can be used
    {
        count++;
        change -= dime;
    }

    while (change >= nickel) //check if the nickel can be used
    {
        count++;
        change -= nickel;
    }

    while (change >= penny) //check if the penny can be used
    {
        count++;
        change -= penny;
    }

    printf("Number of coins: %i\n", count); //print the count

}

File: pset6/cash/cash.py
# cash.py

# Brandan McDevitt
# Harvard Computer Science 50
# Problem Set 6

# Port cash.c to python.

from cs50 import get_float

quarter = 25
dime = 10
nickel = 5
penny = 1
count = 0

while True:
    change = get_float("Enter change: ")

    if change > 0:
        break

change *= 100
change = round(change)

while change >= quarter:
    count += 1
    change -= quarter

while change >= dime:
    count += 1
    change -= dime

while change >= nickel:
    count += 1
    change -= nickel

while change >= penny:
    count += 1
    change -= penny

print(count)